src/text/text.cpp

```diff
@@ -33,71 +39,84 @@ namespace text {
 		auto start = (file_size != 0 && file_content[0] == '#') ? parsers::csv_advance_to_next_line(file_content, file_content + file_size) : file_content;
 		while(start < file_content + file_size) {
 			start = parsers::parse_first_and_nth_csv_values(language, start, file_content + file_size, ';', [&state](std::string_view key, std::string_view content) {
-
-				char const* section_start = content.data();
 				char const* seq_start = content.data();
 				char const* seq_end = content.data() + content.size();
+				char const* section_start = seq_start;
 
 				const auto component_start_index = state.text_components.size();
-
-				for(const char* pos = seq_start; pos < seq_end; ++pos) {
+				for(char const* pos = seq_start; pos < seq_end; ) {
+					bool colour_esc = false;
 					if(uint8_t(*pos) == 0xA7) {
 						if(section_start != pos) {
 							auto added_key = state.add_to_pool(std::string_view(section_start, pos - section_start));
 							state.text_components.emplace_back( added_key );
 						}
-						section_start = pos + 2;
-						if(pos + 1 < seq_end)
-							state.text_components.emplace_back( char_to_color(*(pos + 1)) );
-						pos += 1;
+						section_start = pos += 1;
```

Comment:
This sort of thing (doing assignment and then using the return value) is a real bad habit. I know it may sound petty, but it is. Break it up into two lines (and in the other places as well)